rskj-core/src/main/java/co/rsk/peg/LegacyErpFederation.java

```diff
@@ -0,0 +1,86 @@
+package co.rsk.peg;
+
+import co.rsk.bitcoinj.core.BtcECKey;
+import co.rsk.bitcoinj.core.NetworkParameters;
+import co.rsk.bitcoinj.script.ErpFederationRedeemScriptParser;
+import co.rsk.bitcoinj.script.Script;
+import co.rsk.bitcoinj.script.ScriptBuilder;
+import java.time.Instant;
+import java.util.List;
+import org.ethereum.config.blockchain.upgrades.ActivationConfig;
+import org.ethereum.config.blockchain.upgrades.ConsensusRule;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Deprecated
+public class LegacyErpFederation extends ErpFederation {
+    private static final Logger logger = LoggerFactory.getLogger(LegacyErpFederation.class);
+
+    public LegacyErpFederation(
+        List<FederationMember> members,
+        Instant creationTime,
+        long creationBlockNumber,
+        NetworkParameters btcParams,
+        List<BtcECKey> erpPubKeys,
+        long activationDelay,
+        ActivationConfig.ForBlock activations) {
+
+        super(members, creationTime, creationBlockNumber, btcParams, erpPubKeys, activationDelay, activations);
+    }
+
+    @Override
+    public Script getRedeemScript() {
+        if (!activations.isActive(ConsensusRule.RSKIP284) &&
+            btcParams.getId().equals(NetworkParameters.ID_TESTNET)) {
+            logger.debug("[getRedeemScript] Returning hardcoded redeem script");
+            return new Script(ERP_TESTNET_REDEEM_SCRIPT_BYTES);
+        }
+
+        if (redeemScript == null) {
+            logger.debug("[getRedeemScript] Creating the redeem script from the keys");
+            redeemScript = ErpFederationRedeemScriptParser.createErpRedeemScriptDeprecated(
+                    ScriptBuilder.createRedeemScript(getNumberOfSignaturesRequired(), getBtcPublicKeys()),
+                    ScriptBuilder.createRedeemScript(erpPubKeys.size() / 2 + 1, erpPubKeys),
+                    activationDelay
+                );
+        }
+
+        return redeemScript;
+    }
+
+    @Override
+    public Script getStandardRedeemScript() {
+        if (standardRedeemScript == null) {
+            // i think is not quite right to just "fix" the standardRedeemScript in the buggy fed.
+            // the real case is that the OP_CHECKMULTISIG is not before the OP_ELSE... as the method adds.
```

Comment:
Didn't quite get this

Reply:
My concern is that in the ErpFederationRedeemScriptParser.extractStandardRedeemScript, we are adding the OP_CHECKMULTISIG opcode at the end of the redeem script's chunks

Reply:
The buggy redeem script (used in this class) has the following format approx, I may be missing some op code

```
OP_NOTIF
  OP_M
  pubkey1
  pubkey2
  ...
  pubkeyN
  OP_N
OP_ELSE
OP_CSV
OP_DROP
  OP_M
  emergencyPubkey1
  emergencyPubkey2
  ....
  emergencyPubkeyN
  OP_N
OP_ENDIF
OP_CHECKMULTISIG
```

When extracting the standard redeem script, it gets the chunks between OP_NOTIF and OP_ELSE. So it gets
```
OP_M
  pubkey1
  pubkey2
  ...
  pubkeyN
  OP_N
```
It's missing the OP_CHECKMULTISIG, that's why it needs to add it. Not related to the fact that's it's buggy, that's another story

Reply:
oookidoki. yeah i understood what it does, but i just didnt know why. thanks